python/python/lancedb/table.py

```diff
@@ -73,6 +74,22 @@
 QueryType = Literal["vector", "fts", "hybrid", "auto"]
 
 
+def _schema_without_embedding_funcs(schema: Optional[pa.Schema]) -> Optional[pa.Schema]:
+    """Return a schema without any embedding function columns"""
+    if schema is None:
+        return None
+    embedding_metadata = (
```

Comment:
there's a `.embedding_functions` property on the table i believe. So you don't need to duplicate json decoding

Reply:
Changed to use `EmbeddingFunctionRegistry.parse_functions()`, because the Table might not exist yet.